tests/performance_comparison.py

```diff
@@ -0,0 +1,164 @@
+"""
+This script is designed to analyse the performance of two different test
+runner functions on a set of GitHub repositories. It utilises multiple
+command-line parameters to manage input and output data paths and conducts
+performance analysis using specified test runner functions.
+
+The script processes a CSV file containing repository URLs, applies the test
+runner functions to measure performance metrics such as execution time, memory
+usage, and CPU usage, and then saves these metrics into separate CSV files.
+Additionally, it combines all the metrics into a single CSV for a comprehensive
+view.
+
+Returns:
+    pd.DataFrame: DataFrame containing performance metrics for each repository.
+
+Example Usage:
+    python script_name.py --input-file data/repositories.csv
+    --clone-dir path/to/clone --output-file1 path/to/output1.csv
+    --output-file2 path/to/output2.csv
+    --output-file3 path/to/combined_output.csv
+"""
+
+import argparse
+from pathlib import Path
+
+import pandas as pd
+from loguru import logger
+
+from src.github_repo_request_local import detect_test_runners, detect_test_runners2
+from utils.git_utils import get_working_directory_or_git_root
+from utils.measure_performance import measure_performance, performance_records
+from utils.string_utils import sanitise_directory_name
+
+
+def parse_args():
+    """Parse command line arguments for excluded extensions and clone
```

Comment:
This docstring looks incorrect - was it copy+pasted from another file? There aren't any excluded extensions here :)

Reply:
I did  copy this from the other script. I fixed this.